modules/sov-ibc/src/rpc/context.rs

```diff
@@ -1,30 +1,345 @@
+    {
+        let seq_ack_path = &SeqAckPath::new(port_id, channel_id);
+
+        SV::value_at_key(
+            seq_ack_path,
+            &self.ibc.ack_sequence_map,
+            *self.working_set.borrow_mut(),
+        )
+    }
+
+    pub fn dyn_packet_commitment<SV>(
+        &self,
+        port_id: &PortId,
+        channel_id: &ChannelId,
+        sequence: Sequence,
+    ) -> SV::Output
+    where
+        SV: StorageValue<PacketCommitment>,
+    {
+        let commitment_path = &CommitmentPath::new(port_id, channel_id, sequence);
+
+        SV::value_at_key(
+            commitment_path,
+            &self.ibc.packet_commitment_map,
+            *self.working_set.borrow_mut(),
+        )
+    }
+
+    pub fn dyn_packet_receipt<SV>(
+        &self,
+        port_id: &PortId,
+        channel_id: &ChannelId,
+        sequence: Sequence,
+    ) -> SV::Output
+    where
+        SV: StorageValue<Receipt>,
+    {
+        let receipt_path = &ReceiptPath::new(port_id, channel_id, sequence);
+
+        SV::value_at_key(
+            receipt_path,
+            &self.ibc.packet_receipt_map,
+            *self.working_set.borrow_mut(),
+        )
+    }
+
+    pub fn dyn_packet_acknowledgement<SV>(
+        &self,
+        port_id: &PortId,
+        channel_id: &ChannelId,
+        sequence: Sequence,
+    ) -> SV::Output
+    where
+        SV: StorageValue<AcknowledgementCommitment>,
+    {
+        let ack_path = &AckPath::new(port_id, channel_id, sequence);
+
+        SV::value_at_key(
+            ack_path,
+            &self.ibc.packet_ack_map,
+            *self.working_set.borrow_mut(),
+        )
+    }
+}
 
 impl<'a, S> ProvableContext for IbcContext<'a, S>
 where
     S: Spec,
 {
-    /// TODO: Should figure out how can access the proof from the context
-    fn get_proof(&self, _height: Height, _path: &Path) -> Option<Vec<u8>> {
-        Some(vec![])
+    // NOTE: This is not efficient if used with a separate `get` call which returns only values.
+    // Because `get_with_proof` will retrieve proof along with the value anyway.
+    // Currently there is no way of retrieving only the proof.
+    fn get_proof(&self, height: Height, path: &Path) -> Option<Vec<u8>> {
+        let mut archival_working_set = self
+            .working_set
+            .borrow()
+            .get_archival_at(height.revision_height());
+
+        match path {
+            Path::ClientState(client_state_path) => self
+                .ibc
+                .client_state_map
+                .get_with_proof(&client_state_path.0, &mut archival_working_set)
+                .proof
+                .try_to_vec()
+                .ok(),
+            Path::ClientConsensusState(client_consensus_state_path) => self
+                .ibc
+                .consensus_state_map
+                .get_with_proof(client_consensus_state_path, &mut archival_working_set)
+                .proof
+                .try_to_vec()
+                .ok(),
```

Comment:
I think we can minimize this recurring by calling `.proof.try_to_vec().ok()` after the `match` return.

Reply:
7c5789c